Modules/_lsprof.c

```diff
@@ -780,34 +780,48 @@ _lsprof_Profiler_enable_impl(ProfilerObject *self, int subcalls,
         return NULL;
     }
 
-    if (PyObject_CallMethod(monitoring, "use_tool_id", "is", self->tool_id, "cProfile") == NULL) {
+    PyObject *check = PyObject_CallMethod(monitoring,
+                                          "use_tool_id", "is",
+                                          self->tool_id, "cProfile");
+    if (check == NULL) {
+        PyErr_Clear();
```

Comment:
I don't think that it's useful. It's very common to override a current exception with a new exception in C like that.